clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -875,8 +872,12 @@ private void handleClientResponse(final ClientResponse response,
 
         abstract void onResponse(final ClientResponse response);
 
-        boolean retryTimeoutExpired(long currentTimeMs) {
-            return expirationTimeMs.isPresent() && expirationTimeMs.get() <= currentTimeMs;
+        /**
+         * If at least one attempt has been sent, and the user-provided timeout has elapsed, consider the
+         * request as expired.
+         */
+        boolean retryTimeoutExpired() {
```

Comment:
name nit: this is not only about timeout expired anymore, it considers if the request was sent (significant conceptual diff I would say), so what about aligning the name? (`sentAndExpired` maybe, or whatever you prefer to better reflect what it actually checks now)

Reply:
I solved it by removing the method altogether 😉

Reply:
...and folding the checks into the caller directly.